src/pip/_internal/req/constructors.py

```diff
@@ -334,6 +334,11 @@ def install_req_from_req_string(
             "%s depends on %s " % (comes_from.name, req)
         )
 
+    if req.url:
+        return install_req_from_line(
+            req.url, comes_from=comes_from, isolated=isolated,
+            wheel_cache=wheel_cache
```

Comment:
Also, is there a reason why `use_pep517` is passed on below, but not above? (Something else to test in a unit test of `install_req_from_req_string()`?)

Reply:
Good catch. I think the original patch (https://github.com/pypa/pip/issues/5780#issuecomment-421092322) predated `use_pep517`